python/spacewalk/satellite_tools/download.py

```diff
@@ -282,6 +283,10 @@ def run(self):
                 # log_obj must be thread-safe
                 self.parent.log_obj.log(success, os.path.basename(params['relative_path']))
             self.queue.task_done()
+            if not success:
+                package = os.path.basename(params['relative_path'])
+                # append package name without extension
+                self.failed_pkgs.add(package.rsplit('.', maxsplit=1)[0])
```

Comment:
It makes me a bit confused. Actually afterwards it's checking with NVREA (strange by itself as usually it's called NEVRA as Epoch has the higher priority over the Version, but anywat) the potential problem here, that some of the third party repos are not folliwing the rule to call the package files with this naming convention. For exmple: https://rpm.opera.com/rpm. It contains `opera_stable-103.0.4928.26-linux-release-x64-signed.rpm` with `NEVRA`=`google-chrome-stable-118.0.5993.70-1.x86_64`

Reply:
Nice, makes sense! Now, instead of relying on the package naming rules, I rely on paths. This means that there's no need for removing the extension anymore, and from my understanding, the target path should always be defined.

WDYT about the current version?

Reply:
Thanks for fixing it :+1: Looks good for me.